users/luescher/setups/hybrid/gmm_system.py

```diff
@@ -0,0 +1,845 @@
+
+        for name, v in sorted(test_data.items()):
+            self.corpora[name] = v.corpus_object
+            self.concurrent[name] = v.concurrent
+            self._init_corpora(name)
+            self._init_lexica(name, **v.lexicon)
+            self._init_lm(name, **v.lm)
+            self.test_corpora.append(name)
+        self.cart_questions = gmm_args.cart_questions
+
+    @tk.block()
+    def _init_am(self, **kwargs):
+        self.crp["base"].acoustic_model_config = am.acoustic_model_config(**kwargs)
+
+    @tk.block()
+    def _init_corpora(self, name):
+        segm_corpus_job = corpus_recipes.SegmentCorpusJob(
+            self.corpora[name].corpus_file, self.concurrent[name]
+        )
+        self.set_corpus(
+            name=name,
+            corpus=self.corpora[name],
+            concurrent=self.concurrent[name],
+            segment_path=segm_corpus_job.segment_path,
+        )
+        self.jobs[name]["segment_corpus"] = segm_corpus_job
+
+    @tk.block()
+    def _init_lm(self, name, file, type, scale, **kwargs):
+        self.crp[name].language_model_config = rasr.RasrConfig()
+        self.crp[name].language_model_config.type = type
+        self.crp[name].language_model_config.file = file
+        self.crp[name].language_model_config.scale = scale
+
+    @tk.block()
+    def _init_lexica(self, name, file, normalize_pronunciation, **kwargs):
+        self.crp[name].lexicon_config = rasr.RasrConfig()
+        self.crp[name].lexicon_config.file = file
+        self.crp[name].lexicon_config.normalize_pronunciation = normalize_pronunciation
+
+    def allow_zero_weights(self, name: str = "base"):
+        if self.crp[name].acoustic_model_post_config is None:
+            self.crp[name].acoustic_model_post_config = rasr.RasrConfig()
+        self.crp[name].acoustic_model_post_config.mixture_set.allow_zero_weights = True
+        self.crp[
+            name
+        ].acoustic_model_post_config.old_mixture_set.allow_zero_weights = True
+
+    def _add_features(self, name, corpus, feature_job, prefix=""):
+        self.jobs[corpus][name] = feature_job
+        feature_job.add_alias(f"{prefix}{corpus}_{name}_features")
+        self.feature_caches[corpus][name] = feature_job.out_feature_path[name]
+        self.feature_bundles[corpus][name] = feature_job.out_feature_bundle[name]
+        feature_path = rasr.FlagDependentFlowAttribute(
+            "cache_mode",
+            {
+                "task_dependent": self.feature_caches[corpus][name],
+                "bundle": self.feature_bundles[corpus][name],
+            },
+        )
+        self.feature_flows[corpus][name] = features.basic_cache_flow(feature_path)
+        self.feature_flows[corpus][f"uncached_{name}"] = feature_job.feature_flow
+
+    @tk.block()
+    def extract_features(self, feat_args: dict, **kwargs):
+        corpus_list = self.train_corpora + self.dev_corpora + self.test_corpora
+
+        if "mfcc" in feat_args.keys():
```

Comment:
I thought we always use the "+energy" in the GMM Pipeline when using MFCC, that is why I did not separate. This is more a query than an objection. I think it would be good to handle this consistent throughout the different SystemClasses (GMM, HybridNN, etc)